lib/smart_proxy_ansible/plugin.rb

```diff
@@ -19,6 +19,8 @@ class Plugin < Proxy::Plugin
         require 'smart_proxy_ansible/runner/command_creator'
         require 'smart_proxy_ansible/runner/playbook'
 
+        raise NotExistingWorkingDirException.new("Working Directory does not exist") unless File.directory? Proxy::Ansible::Plugin.settings[:working_dir] || Proxy::Ansible::Plugin.settings[:working_dir].nil?
```

Comment:
The parts of the condition should probably be switched around. If the setting is `nil`, then the first check will raise an exception

```
[1] pry(main)> File.directory? nil
TypeError: no implicit conversion of nil into String
from (pry):1:in `directory?'
```

Reply:
Sorry, it is a draft haven't checked it yet. what do you mean by switching them around? 
like this :
Proxy::Ansible::Plugin.settings[:working_dir].nil? || File.directory? Proxy::Ansible::Plugin.settings[:working_dir].nil?  ?

Reply:
Just switching the order so instead of
```ruby
unless File.directory? Proxy::Ansible::Plugin.settings[:working_dir] || Proxy::Ansible::Plugin.settings[:working_dir].nil?
```

you would have

```ruby
unless Proxy::Ansible::Plugin.settings[:working_dir].nil? || File.directory? Proxy::Ansible::Plugin.settings[:working_dir]
```

Reply:
@adamruzicka  I am trying to check it now, and I see that when the exception is inside that after_actvation block the job is marked as a success. It shouldn't be before the after_activiation block?